aten/src/ATen/native/cpu/TensorCompareKernel.cpp

```diff
@@ -161,10 +161,76 @@ static void where_kernel_impl(TensorIterator &iter, ScalarType condition_type) {
   });
 }
 
+static void isposinf_kernel_for_bool_impl(TensorIterator& iter) {
```

Comment:
These kernels look correct, but do we need them? That is, how would this PR looked if you acquired std::numeric_limits<scalar_t>::infinity() and tested for equality with it using the `torch.equal` function, for example?

Reply:
Thanks for pointing this out, the kernel implementations will remain as I am more familiar with `TensorIterator` at this moment.